### PR TITLE
Build number in tag

### PR DESCRIPTION
**Pull request by M0ses**
The first commit includes the significant code changes
The second is only cleanup to stop rubocop noise

### CODE REVIEW

File: lib/sle2docker/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Sle2Docker
-  VERSION = '0.4.2'.freeze
+  VERSION = '0.5.0'.freeze
```
Comment (mssola) [inline lib/sle2docker/version.rb]: Why ? This should go into a separate PR, with an update on the changelog and all that. Please, remove this change.

**Comment by M0ses** [comment]
I thought it would be a good idea - but this was, before I made new behavior optional. Reverted it

**Comment by M0ses** [comment]
@mssola Added the tests as discussed on ML. Please review again.

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,6 @@ pkg
 Vagrantfile
 package/sle2docker.8
 package/sle2docker.8.html
+*.sw?
+sle2docker-*.gem
+Gemfile.lock
```
Comment (mssola) [inline .gitignore]: A couple of things :grin: 

- What is `*.sw` ? If it isn't strictly coupled with this gem, please remove it.
- Don't add the Gemfile.lock here. We should upload a proper one before releasing the next version.

File: test/prebuilt_image_test.rb [hunk 3]
```diff
@@ -40,7 +40,17 @@ class PrebuiltImageTest < MiniTest::Test
         end
       end
     end
+  end
+end
+class PrebuiltImageTest < MiniTest::Test
```
Comment (mssola) [inline test/prebuilt_image_test.rb]: Add a separation line between these two classes.

File: test/prebuilt_image_test.rb [hunk 4]
```diff
@@ -86,6 +96,52 @@ class PrebuiltImageTest < MiniTest::Test
 
         prebuilt_image.activate
       end
+      it 'triggers docker build and tags with build #' do
```
Comment (mssola) [inline test/prebuilt_image_test.rb]: Add a separation line between each `it` case.

File: test/prebuilt_image_test.rb [hunk 5]
```diff
@@ -86,6 +96,52 @@ class PrebuiltImageTest < MiniTest::Test
...
+          @options
+        )
+        prebuilt_image.expects(:prepare_docker_build_root).once.returns(tmp_dir)
+        prebuilt_image.expects(:verify_image).once
+        Docker::Image.expects(:build_from_dir).with(tmp_dir).once.returns(mocked_image)
+        FileUtils.expects(:rm_rf).with(tmp_dir).once
+
+        prebuilt_image.activate
+      end
+      it 'triggers docker build and tags with build# (build# empty)' do
```
Comment (mssola) [inline test/prebuilt_image_test.rb]: Same here

**Comment by mssola** [comment]
After fixing these nitpicks, could you squash all your commits into a single one and sign it off (the `-s` option of `git commit`). Thanks :clap:

File: .gitignore (hunk 2 above)
Comment (M0ses) [inline .gitignore]: .*sw? - vim swap files - will remove it

**Comment by mssola** [comment]
Thanks :clap:
